core/src/main/java/org/mule/runtime/core/internal/construct/AbstractFlowConstruct.java

```diff
@@ -74,6 +78,9 @@ public abstract class AbstractFlowConstruct extends AbstractExecutableComponent
   private volatile FlowConstructStatistics statistics;
   protected FlowStoppedPersistenceListener flowStoppedPersistenceListener;
 
+  @Inject
+  private FeatureFlaggingService featureFlaggingService;
```

Comment:
Are we sure this is injected ok like this in every case? Have you tested the change in a mule runtime?
It seems strange to me that the MuleContext is injected by constructor programmatically.

Reply:
It is injected correctly in a standalone. It seems it is not injected in some tests where the muleContext is mocked but we can verify that in the refactor task.